src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/roi_align.cpp

```diff
@@ -19,44 +19,88 @@ const std::vector<InferenceEngine::Precision> netPRCs = {
-
-const auto ROIAlignCases_max =
-    ::testing::Combine(
-                       ::testing::ValuesIn(
-                                           std::vector<std::vector<size_t>> {
-                                               { 2, 8, 20, 20 },
-                                               { 2, 1, 20, 20 },
-                                               { 2, 1, 10, 20 }
-                                           }),
-                       ::testing::Values(std::vector<size_t>{ 2, 4 }),
-                       ::testing::Values(2),
-                       ::testing::Values(2),
-                       ::testing::ValuesIn(std::vector<float> { 1, 0.625 }),
-                       ::testing::Values(2),
-                       ::testing::Values("max"),
-                       ::testing::ValuesIn(netPRCs),
-                       ::testing::Values(CommonTestUtils::DEVICE_GPU)
-);
-
-INSTANTIATE_TEST_SUITE_P(smoke_TestsROIAlign_max, ROIAlignLayerTest, ROIAlignCases_max, ROIAlignLayerTest::getTestCaseName);
+const auto ROIAlignCases_average = ::testing::Combine(
+    ::testing::ValuesIn(std::vector<std::vector<size_t>>{{3, 8, 16, 16}, {2, 1, 16, 16}, {2, 1, 8, 16}}),
+    ::testing::Values(std::vector<size_t>{2, 4}),
+    ::testing::Values(2),
+    ::testing::Values(2),
+    ::testing::ValuesIn(std::vector<float>{1, 0.625}),
+    ::testing::Values(2),
+    ::testing::Values("avg"),
+    ::testing::ValuesIn(netPRCs),
+    ::testing::Values(CommonTestUtils::DEVICE_GPU));
+
+INSTANTIATE_TEST_SUITE_P(smoke_TestsROIAlign_average,
+                         ROIAlignLayerTest,
+                         ROIAlignCases_average,
+                         ROIAlignLayerTest::getTestCaseName);
+
+const auto ROIAlignCases_max = ::testing::Combine(
+    ::testing::ValuesIn(std::vector<std::vector<size_t>>{{2, 8, 20, 20}, {2, 1, 20, 20}, {2, 1, 10, 20}}),
+    ::testing::Values(std::vector<size_t>{2, 4}),
+    ::testing::Values(2),
+    ::testing::Values(2),
+    ::testing::ValuesIn(std::vector<float>{1, 0.625}),
+    ::testing::Values(2),
+    ::testing::Values("max"),
+    ::testing::ValuesIn(netPRCs),
+    ::testing::Values(CommonTestUtils::DEVICE_GPU));
+
+INSTANTIATE_TEST_SUITE_P(smoke_TestsROIAlign_max,
+                         ROIAlignLayerTest,
+                         ROIAlignCases_max,
+                         ROIAlignLayerTest::getTestCaseName);
+
+const auto ROIAlignCases_average_asymmetric = ::testing::Combine(
+    ::testing::ValuesIn(std::vector<std::vector<size_t>>{{2, 1, 8, 8}, {2, 8, 20, 20}, {2, 1, 20, 20}, {2, 1, 10, 20}}),
+    ::testing::Values(std::vector<size_t>{2, 4}),
+    ::testing::Values(2),
+    ::testing::Values(2),
+    ::testing::ValuesIn(std::vector<float>{1, 0.625}),
+    ::testing::Values(2),
+    ::testing::Values("avg"),
+    ::testing::Values("asymmetric"),
+    ::testing::ValuesIn(netPRCs),
+    ::testing::Values(CommonTestUtils::DEVICE_GPU));
+
+INSTANTIATE_TEST_SUITE_P(smoke_TestsROIAlign_avg_asym,
```

Comment:
Can we keep  ROIAlign-3 test to check backward compatibility?

Reply:
actually we do keep them, we have two ROIAlign-3 tests along with a new ones (smoke_TestsROIAlign_average and smoke_TestsROIAlign_max)